docs/architecture.md

```diff
@@ -1,42 +1,104 @@
 
-## OCI Image based
+### {{elemental.cli.name}}
+{{elemental.cli.name}} is the tool that allows to turn the Elemental OS image in a bootable and installed OS: it can generate an {{elemental.iso.name}} image from the provided Elemental OS container image. The generated {{elemental.iso.name}} image can be used to boot a virtual machine or a bare metal host and start the Elemental OS installation.
 
-Elemental is an image based distribution with an A/B style update mechanism. One first runs
-on a read-only image A and to do an upgrade pulls a new read only image
-B and then reboots the system to run on B. What is unique about
-Elemental is that the runtime images come from OCI Images. Not an
-OCI Image containing special artifacts, but an actual Docker runnable
-image that is built using standard Docker build processes. Elemental is
-built using normal `docker build` and if you wish to customize the OS
-image all you need to do is create a new `Dockerfile`.
+The {{elemental.cli.name}} allows also the Elemental OS installation on the storage device of the live booted host, applying the Elemental installation configuration passed in inout. For the list and syntax of the commands available in the {{elemental.cli.name}}, check the [online documentation]({{elemental.cli.url}}).
 
 ## {{elemental.operator.name}}
-
-Elemental includes no container runtime, Kubernetes distribution,
-or Rancher itself. All of these assets are dynamically pulled at runtime. All that
-is included in Elemental is [{{elemental.operator.name}}]({{elemental.operator.url}}) which
-is responsible for managing OS upgrades and managing a secure device inventory to assist
+The {{elemental.operator.name}} is responsible for managing OS upgrades and managing a secure device inventory to assist
 with zero touch provisioning.
+It provides an {{elemental.operator.name}} Helm Chart and an {{elemental.register.name}}.
+
+### {{elemental.operator.name}} Helm Chart
+The {{elemental.operator.name}} Helm Chart must be installed on a Rancher Cluster. It enables new hosts to:
+
+- register against the {{elemental.operator.name}}.
+- retrieve the Elemental installation configuration required for the Elemental OS installation from Kubernetes resources.
+- download and install the [{{ranchersystemagent.name}}]({{ranchersystemagent.url}}), which enables Rancher to provision and manage K3s and RKE2 on the Elemental nodes.
+
+The {{elemental.operator.name}} allows control of the Elemental Nodes by extending the Kubernetes APIs with a set of _elemental.cattle.io_ [Kubernetes CRDs](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/):
+
+- MachineRegistration
+- MachineInventory
+- MachineInventorySelector
+- MachineInventorySelectorTemplate
+- ManagedOSImage
+- ManagedOSVersion
+- ManagedOSVersionChannel
+
+#### MachineRegistration
+A MachineRegistration resource should be created for each new host to be provisioned by Elemental.
```

Comment:
My impression was that machine registration can manage a set of hosts

Reply:
yep right, dropping this liine